sage-ios/SAGE/SAGE/ProfileView.swift

```diff
@@ -0,0 +1,224 @@
+    let profileUserImgBorder = UIImageView()
+    let profileEditButton = UILabel()
+    let userName = UILabel()
+    let userVolunteerLevel = UILabel()
+    let userSchool = UILabel()
+    let topBorder = UIView()
+    let bottomBorder = UIView()
+    let centerBorder = UIView()
+    let userStatusContainer = UIView()
+    let userStatusLabel = UILabel()
+    let userCommitmentContainer = UIView()
+    let userCommitmentLabel = UILabel()
+    let logoutButton = UITableViewCell()
+    let logoutButtonLabel = UILabel()
+    let allCheckIns = UITableViewCell()
+    let allCheckInsLabel = UILabel()
+    
+    // profile page constants
+    let viewHeight = CGFloat(351.1)
+    let headerOffset = CGFloat(500)
+    let leftMargin = CGFloat(30)
+    let headerHeight = CGFloat(120)
+    let titleFontSize = CGFloat(22)
+    let profileImageSize = CGFloat(90)
+    let profileImageBorder = CGFloat(6)
+    let boxHeight = CGFloat(100)
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.addSubview(self.profileContent)
+        self.profileContent.addSubview(self.header)
+        self.profileContent.addSubview(self.profileEditButton)
+        self.profileContent.addSubview(self.profileUserImgBorder)
+        self.profileContent.addSubview(self.profileUserImg)
+        self.profileContent.addSubview(self.userName)
+        self.profileContent.addSubview(self.userSchool)
+        self.profileContent.addSubview(self.userVolunteerLevel)
+        self.profileContent.addSubview(self.topBorder)
+        self.profileContent.addSubview(self.bottomBorder)
+        self.profileContent.addSubview(self.centerBorder)
+        self.profileContent.addSubview(self.userStatusContainer)
+        self.userStatusContainer.addSubview(self.userStatusLabel)
+        self.profileContent.addSubview(self.userCommitmentContainer)
+        self.userCommitmentContainer.addSubview(self.userCommitmentLabel)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func setupWithUser(user: User) {
+        self.profileUserImg.setImageWithUser(user)
+        
+        self.userName.text = user.fullName()
+        self.userSchool.text = user.school?.name
+        self.userVolunteerLevel.text = user.volunteerLevelToString(user.level)
+        
+        let boldString = [NSFontAttributeName: UIFont.getSemiboldFont(30)]
+        
+        let hoursCompleted = user.totalHours
+        let userStatusString = NSMutableAttributedString(string: String(hoursCompleted) + " / 60 \nhours") // ask charles about the 60
+        userStatusString.addAttributes(boldString, range: NSRange(location: 0, length: 2))
+        userStatusString.addAttribute(NSForegroundColorAttributeName, value: UIColor.secondaryTextColor, range: NSRange(location: 8, length: 6))
+        self.userStatusLabel.attributedText = userStatusString
+        self.userStatusLabel.sizeToFit()
```

Comment:
this layout stuff may be better left inside the layoutSubviews method. anything related to position or sizing should be in layoutSubviews unless its constant. for constant dimensions, that stuff can go in a setupSubviews method (you should create that). in this setupSubviews method, stuff like view configurations would go in there, like text alignment for a uilabel, or background color for a view. this setup method would be called only once - in the init method